### PR TITLE
Retry the function set_build_status and catch its failure

### PR DESCRIPTION
**Pull request by JennyChenyj**
[skip ci]

Signed-off-by: Jenny Chen <Jenny.Chen@ibm.com>

### CODE REVIEW

**Comment by JennyChenyj** [comment]
Hey @AdamBrousseau! Thanks for reviewing it! :)

File: buildenv/jenkins/common/pipeline-functions.groovy [hunk 1]
```diff
@@ -345,7 +345,11 @@ def build_with_slack(DOWNSTREAM_JOB_NAME, ghprbGhRepository, ghprbActualCommit,
         // Set Github Commit Status
         if (ghprbActualCommit) {
             node(SETUP_LABEL) {
-                set_build_status(GITHUB_REPO, DOWNSTREAM_JOB_NAME, ghprbActualCommit, DOWNSTREAM_JOB_URL, 'SUCCESS', "Build PASSED")
+                try {
+                    retry_and_delay(set_build_status(GITHUB_REPO, DOWNSTREAM_JOB_NAME, ghprbActualCommit, DOWNSTREAM_JOB_URL, 'SUCCESS', "Build PASSED")
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/pipeline-functions.groovy]: I think you need to wrap your command in `{ curly braces }`

Comment (AdamBrousseau) [reply buildenv/jenkins/common/pipeline-functions.groovy]: I also think we should be adding this to all the places where we run `set_build_status`

Comment (JennyChenyj) [reply buildenv/jenkins/common/pipeline-functions.groovy]: Great idea. Thanks for checking!

Comment (JennyChenyj) [reply buildenv/jenkins/common/pipeline-functions.groovy]: Hey @AdamBrousseau! I changed all the function calls. Thanks!

**Comment by AdamBrousseau** [comment]
Jenkins test sanity plinux jdk11

File: buildenv/jenkins/common/pipeline-functions.groovy [hunk 2]
```diff
@@ -286,8 +286,12 @@ def build_with_slack(DOWNSTREAM_JOB_NAME, ghprbGhRepository, ghprbActualCommit,
     // Set Github Commit Status
     if (ghprbActualCommit) {
         node(SETUP_LABEL) {
-            set_build_status(GITHUB_REPO, DOWNSTREAM_JOB_NAME, ghprbActualCommit, BUILD_URL, 'PENDING', "Build Started")
-        }
+            try {
+                retry_and_delay({set_build_status(GITHUB_REPO, DOWNSTREAM_JOB_NAME, ghprbActualCommit, BUILD_URL, 'PENDING', "Build Started")})
+            } catch (e) {
+                println "set_build_status failed"
+            }
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/pipeline-functions.groovy]: Some of these lines have extra whitespace at the end of the line. Please cleanup then this looks good to me.

Comment (JennyChenyj) [reply buildenv/jenkins/common/pipeline-functions.groovy]: Hey @AdamBrousseau! Thanks for checking! I believe it is clean now. Thanks!

File: buildenv/jenkins/common/pipeline-functions.groovy [hunk 3]
```diff
@@ -286,7 +286,11 @@ def build_with_slack(DOWNSTREAM_JOB_NAME, ghprbGhRepository, ghprbActualCommit,
     // Set Github Commit Status
     if (ghprbActualCommit) {
         node(SETUP_LABEL) {
-            set_build_status(GITHUB_REPO, DOWNSTREAM_JOB_NAME, ghprbActualCommit, BUILD_URL, 'PENDING', "Build Started")
+            try {
+                retry_and_delay({set_build_status(GITHUB_REPO, DOWNSTREAM_JOB_NAME, ghprbActualCommit, BUILD_URL, 'PENDING', "Build Started")})
+            } catch (e) {
+                println "set_build_status failed"
```
Comment (pshipton) [inline buildenv/jenkins/common/pipeline-functions.groovy]: Should there be distinct messages for each of these cases, to make it easy to tell which one failed?

Comment (JennyChenyj) [reply buildenv/jenkins/common/pipeline-functions.groovy]: Sure. Thanks for checking! :)

Comment (JennyChenyj) [reply buildenv/jenkins/common/pipeline-functions.groovy]: Hi @pshipton! I just amended the messages. Please let me know if there is any other change. Thanks for checking it again!

**Comment by pshipton** [comment]
jenkins compile plinux jdk8

**Pull request merged**